### ISSUE TITLE
Wrong ambient temperature

### ISSUE THREAD

**Issue by theremotheman** [BotInABoxER/Wanhao-i3-Custom-Firmware#1]
Hey guys, firstly- thanks for your work on getting marlin working on i3. I'm just after flashing latest hex from your GitHub to my duplicator (v 2.1) and I'm afraid that -at least in my case - idle temp readings are off by ~15c(both bed and extruder are reporting 8c ambient temp when I'm sure it's not that cold in my house, should be 18c -25), I tried to heat bed (to 40c) and everything works, but I'm afraid that reported temperatures are off by that 15c margin. Also, the hex file reports firmware version 1.0.1 but as I see the newest is 1.1.1. Thanks in advance

**Comment by theremotheman** [comment]
After looking for some answers - it looks like my melzi board use 4.7k pull up resistor, in https://www.thingiverse.com/thing:2450111 people suggests setting thermistor table to 1, I'm going to check that later today

Edit: I'm almost sure it's 4.7k, i was going to change table to 1 as metioned on thingiverse but your firmware refuse to compile in arduino ide(and platformio) with error:

```
c:/users/xxx/desktop/arduino-1.6.9/hardware/tools/avr/bin/../lib/gcc/avr/4.8.1/../../../../avr/bin/ld.exe: C:\Users\xxx\AppData\Local\Temp\builde17be87b161bdee619a8fa27708d8fc9.tmp/Marlin.ino.elf section `.text' will not fit in region `text'

c:/users/xxx/desktop/arduino-1.6.9/hardware/tools/avr/bin/../lib/gcc/avr/4.8.1/../../../../avr/bin/ld.exe: region `text' overflowed by 2696 bytes

collect2.exe: error: ld returned 1 exit status

exit status 1
```

I tried with ver 1.8.8 and 1.6.9

Edit 2: It was the resistor, compiled my own marlin firmware from branch 2.0 and everything works fine, even compiled in arduinoide 1.8.8

**Comment by mtpiercey** [comment]
Yeah, sorry I didn't notice this issue earlier; glad to see you got it sorted out.

I was pretty sure that there would be issues regarding the temperature - I had the same issues at first but then updated to a different custom thermistor table; ironically an older commit probably contains the right table for your printer but I changed it to fit mine.

Also, very cool that you got 2.0 working on the i3; I personally thought it wasn't possible but then I didn't try it so there.

**Comment by the-redone** [comment]
Sorry to drag up a closed issue, but having this exact same issue, have tried to use table 1, but adruino spits out this `In file included from sketch\thermistortables.h:44:0,

                 from sketch\temperature.h:30,

                 from sketch\stepper.cpp:84:

thermistortable_1.h:25:12: error: 'OV' was not declared in this scope

   { OV(  23), 300 },

            ^

thermistortable_1.h:26:12: error: 'OV' was not declared in this scope

   { OV(  25), 295 },

            ^

thermistortable_1.h:27:12: error: 'OV' was not declared in this scope

   { OV(  27), 290 },`

It continues until the end of the table, selecting anything other than table 99 (default) results in the same

Any way to fix this? 

Not too clued up on this tbh, but notice that the format for table 99 is completely different to the rest.

**Comment by mtpiercey** [comment]
Yeah, so I sorta broke this firmware with the last commit and I don't feel like fixing it; try https://github.com/BotInABoxER/marlin2-for-wanhao-i3. It's using the latest version of Marlin, has some extra features, and actually works the way it's supposed to work. I also re-did Table 99.

Thanks for mentioning this; I'll be shutting down this repo and moving to the new one soon.

**Comment by the-redone** [comment]
Thanks, i'll give that a go over tomorrow, is table 99 using the 4.7k resistor now too?

**Comment by mtpiercey** [comment]
Table 99 is for printers with a 10k pull-up.

**Comment by the-redone** [comment]
Sorry ye, late one for me, was still assuming table 1 was broken, but it's now default, flashed it over just now, just doing a cube test print, all looks good

one niggle, for me, not sure if it's me or not but the x axis seems to go further than before when doing a home all, the limit switch seems to be press more than usual and the nozzle seems to end up in the very left corner, again i'm not sure if it's different to what it was on the original, it looks different to me at least.

Also, boot screen shows marlin 1.0.0 :P

**Comment by mtpiercey** [comment]
Glad to hear it works for you. Yeah, most likely things are a bit different using vanilla Marlin (as opposed to whatever Wanhao did with it - https://github.com/garychen99/i3V2.1). Also, I just changed the firmware version number. I took it to be the custom version number (i.e. the new edits I did to the v2 firmware just got fully-released, hence the 1.0.0, but yeah changing it to 2 makes more sense.)

IMO, as long as I don't hear strange clicking, scraping, zapping, scratching, or thumping noises or smell noxious fumes coming from the printer, I assume everything is fine. Might be a bit of an assumption, but so far it hasn't failed me.
